smif/data_layer/sector_model_config.py

```diff
@@ -4,34 +4,55 @@
 import logging
 import os
 from glob import glob
+
 import fiona
-from . parse_config import ConfigParser
+
+from .load import load
+from .validate import validate_interventions
 
 
 class SectorModelReader(object):
     """Parses the configuration and input data for a sector model
 
     Arguments
     =========
-    model_name : str
-        The name of the model
-    model_path : str
-        The path to the python module file that contains an implementation
-        of SectorModel
-    model_classname : str
-        The name of the class that implements SectorModel
-    model_config_dir : str
-        The root path of model config/data to use
+    initial_config : dict
```

Comment:
Nice docstrings!  The following format makes each of the keys here a mini-heading::
```python
    initial_config: dict
        Sector model details, sufficient to read the full config from a set of
        files. Must contain the following fields:

            "model_name"
                The name of the sector model, for reference within the
                system-of-systems model

            "model_path"
                The path to the python module file that contains an
                implementation of SectorModel

            "model_classname"
                The name of the class that implements SectorModel

            "model_config_dir"
                The root path of model config/data to use, which
                must contain inputs.yaml, outputs.yaml, time_intervals.yaml and
                regions.shp/regions.geojson

            "initial_conditions"
                List of files containing initial conditions

            "interventions"
                List of files containing interventions
```